src/components/clientTable/ClientTable.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Separator } from "@/components/ui/separator";
+
+import { copyText } from "@/lib/utils";
+import useCalls from "@/lib/hooks/useCalls";
+import { CRMEntry } from "@/lib/crm/types";
+
+interface DataTableProps {
+  columns: ColumnDef<CRMEntry>[];
+  data: CRMEntry[];
+}
+
+export function ClientTable({ columns, data }: DataTableProps) {
+  const [globalFilter, setGlobalFilter] = useState("");
+  const { data: session } = useSession();
+
+  const { makeCall } = useCalls({
+    email: session?.user?.email ?? "",
+    workerSid: "WK3b277b4e6a1d67f2240477fa33f75ea4", // TODO link twilio worker id with okta
+    friendlyName: session?.user?.name ?? "",
+  });
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    onGlobalFilterChange: setGlobalFilter,
+    getFilteredRowModel: getFilteredRowModel(),
+    state: {
+      globalFilter,
+    },
+    meta: {
+      onPhoneNumberCopy: (number: string) => {
+        copyText(number);
+      },
+      onMakeCall: (number: string) => {
+        makeCall(number);
+      },
+    },
+  });
+
+  return (
+    <div>
+      <div className="grid gap-6">
+        <div className="grid gap-2 pt-4">
+          <div className="grid w-full max-w-sm items-center gap-1.5">
+            <Label>Search</Label>
+            <Input
+              id="clientSearch"
+              placeholder="Name, phone number, email"
+              value={globalFilter ?? ""}
+              onChange={(e) => setGlobalFilter(String(e.target.value))}
+            />
+          </div>
+          <p className="italic text-secondary-foreground text-sm">
+            {data.length} clients synced from CRM
+          </p>
+          <Separator />
+        </div>
+        <div className="rounded-md border bg-white">
+          <Table>
+            <TableHeader>
+              {table.getHeaderGroups().map((headerGroup) => (
+                <TableRow key={headerGroup.id}>
+                  {headerGroup.headers.map((header) => {
+                    return (
+                      <TableHead key={header.id}>
+                        {header.isPlaceholder
+                          ? null
+                          : flexRender(
+                              header.column.columnDef.header,
+                              header.getContext()
+                            )}
+                      </TableHead>
+                    );
+                  })}
+                </TableRow>
+              ))}
+            </TableHeader>
+            <TableBody>
+              {table.getRowModel().rows?.length ? (
+                table.getRowModel().rows.map((row) => (
```

Comment:
Minor performance improvement: the `table.getRowModel().rows` call should be done once outside the render (above `return`) and defined as a separate variable. This also avoids any possible race conditions.